# Gather summary stats for read alignments

# There are several ways to count the reads in the raw and trimmed files, such as:
#   - Using wc -l to get line count (but then need to divide by 4)
#   - Using grep to find specific symbols (like lines starting with @, indicating the start of a sequence name)
#   - Using the count data already generated by FASTQC/MultiQC
#  For simplicity, I'm going to do the first option (using wc -l)

# Load Samtools to get stats out of the alignment file
module load SAMtools/1.18-GCC-12.3.0

# Set up a tidy-formtted data file with 3 columns: sample, step, and count
#    NOTE: You could set up a wider-formatted one wtih each row being a sample and each column being a step 
#          (Raw, Trimmed, Aligned, Variants), but that requires some extra manipulation in R
report=results/summary_stats.csv
echo -e "sample,step,count" > $report


# Loop over samples to get counts
for fwd in data/raw_fastq/*_1.fastq.gz; do
    sample=$(basename $fwd _1.fastq.gz)
    echo "Getting stats for sample $sample"
    
    # Raw count
    rawcount=$(zcat $fwd | wc -l)
    rawcount=$(($rawcount/4))  # Have to divide by 4 to get actual count; we didn't cover in class, so need to look up
    echo "$sample,raw,$rawcount" >> $report
    
    # Trimmed count
    trimmed=data/trimmed_fastq/${sample}_1.paired.fastq.gz  # Trimmed, paired forward reads
    trimcount=$(zcat $trimmed | wc -l)
    trimcount=$(($trimcount/4))  # Have to divide by 4 to get actual count; we didn't cover in class, so need to look up
    echo "$sample,trimmed,$trimcount" >> $report

    # Aligned count; can do the raw SAM, raw BAM, or sorted BAM; all should be the same.
    #    I prefer sorted because that's what I'll be using later
    # Again, there are several ways to do this. I prefer using the --count option so Samtools counts for me
    #    The -F 0x04 flag filters out unaligned reads
    #    Another option is "samtools stats", though you then have to capture that and write it to a file.
    aligned=results/bam/$sample.sorted.bam
    aligncount=$(samtools view -F 0x04 --count $aligned)
    echo "$sample,aligned,$aligncount" >> $report
    
    
    # Variants; easiest way to do this is to count the non-header lines
    #   Here I let grep do the counting again. "-v" means "take what doesn't match", and "^#" means 
    #   the line starts with a hashtag, so it's only counting lines that do NOT start with hashtags
    variants=results/vcf/$sample.vcf
    varcount=$(grep -v "^#" -c $variants) 
    echo "$sample,variants,$varcount" >> $report
    
    
done

